src/lamp_py/bus_performance_manager/tm_ingestion.py

```diff
@@ -0,0 +1,302 @@
+from typing import List
+
+import pytz
+import polars as pl
+
+from lamp_py.runtime_utils.remote_files import RemoteFileLocations
+
+BOSTON_TZ = pytz.timezone("EST5EDT")
+UTC_TZ = pytz.utc
+
+
+def create_dt_from_sam(
+    service_date_col: pl.Expr, sam_time_col: pl.Expr
+) -> pl.Expr:
+    """
+    add a seconds after midnight to a service date to create a datetime object.
+    seconds after midnight is in boston local time, convert it to utc.
+    """
+    return (
+        service_date_col.cast(pl.Datetime) + pl.duration(seconds=sam_time_col)
+    ).map_elements(
+        lambda x: BOSTON_TZ.localize(x).astimezone(UTC_TZ),
+        return_dtype=pl.Datetime,
+    )
+
+
+def generate_tm_events(tm_files: List[str]) -> pl.DataFrame:
```

Comment:
this will generate the tm vehicle events dataframe that can be joined with the gtfs rt dataframe to create contain all vehicle events we're aware of.